### ISSUE TITLE
Weird results when use Segmentation Pass for inference

### ISSUE THREAD

**Issue by luuil** [PeterL1n/RobustVideoMatting#123]
https://github.com/PeterL1n/RobustVideoMatting/blob/f8a26e27198a93a94bfd06e96b8d5a34d0660f81/inference.py#L127

I changed this line to use Segmentation Pass. (use the pretrained weights `rvm_mobilenetv3.pth`)

```python
pha, *rec = model(src, *rec, segmentation_pass=True)
fgr = src * pha
```

But I got weird mask results, something like this, why?

![seg_pass_alpha](https://user-images.githubusercontent.com/26055346/148023232-60dd6715-5307-4623-ac3f-1f1d9bd2ff71.jpg)

**Comment by PeterL1n** [comment]
Try `pha.sigmoid()`, aka `fgr = src * pha.sigmoid()`. Because the segmentation pass outputs logits. You need a sigmoid to convert it to 0-1 range.

**Comment by luuil** [comment]
Thank you for your prompt reply. You are right, now the result is correct.

**Comment by FengMu1995** [comment]
is it necessary to  change "pha.sigmoid()" for the train module
"""
pred_seg = self.model_ddp(true_img, segmentation_pass=True)[0]
 loss = segmentation_loss(pred_seg, true_seg)
"""

**Comment by vacing** [comment]
> 



> is it necessary to change "pha.sigmoid()" for the train module


it seems not necessary

https://zhang-yang.medium.com/how-is-pytorchs-binary-cross-entropy-with-logits-function-related-to-sigmoid-and-d3bd8fb080e7
